File: lambda-api/MyService/DevService.ts
import {BaseMessage as $UniRpc_BaseMessage} from "../UniRpc/BaseMessage";
import {ReturnMessage as $UniRpc_ReturnMessage} from "../UniRpc/ReturnMessage";
import {WebsocketServiceBase as $UniRpc_WebsocketServiceBase} from "../UniRpc/WebsocketServiceBase";
import {Message as $MyService_Message} from "./Message";
export abstract class DevService<Dog> extends $UniRpc_WebsocketServiceBase
{
    public constructor () {
        super();
        this.__reflection = "MyService.DevService";
    }
    public abstract Send(message: $MyService_Message): Promise<boolean>;
    public abstract Get(): Promise<boolean>;
    public abstract StoryCount(): Promise<number>;
    public abstract Test(): Promise<string>;
    public abstract AB(my: number): Promise<number>;
    public abstract Gap(value: number, enabled: boolean): Promise<number[]>;
    /** release resource */
    public abstract Release<Cat>(/** the cat instance */ cat: Cat, dog: Dog): Promise<string>;
    public abstract GetMessage(): Promise<$MyService_Message>;
    public abstract IncomingCall(): Promise<void>;
    public async __invoke(message: $UniRpc_BaseMessage): Promise<$UniRpc_BaseMessage> {
        switch (message.Method)
        {
            case "Send":
                {
                    let ____message: $MyService_Message = message.Payload['message'];
                    return $UniRpc_ReturnMessage(message, await this.Send(____message));
                }
            case "Get":
                {
                    return $UniRpc_ReturnMessage(message, await this.Get());
                }
            case "StoryCount":
                {
                    return $UniRpc_ReturnMessage(message, await this.StoryCount());
                }
            case "Test":
                {
                    return $UniRpc_ReturnMessage(message, await this.Test());
                }
            case "AB":
                {
                    let ____my: number = message.Payload['my'];
                    return $UniRpc_ReturnMessage(message, await this.AB(____my));
                }
            case "Gap":
                {
                    let ____value: number = message.Payload['value'];
                    let ____enabled: boolean = message.Payload['enabled'];
                    return $UniRpc_ReturnMessage(message, await this.Gap(____value, ____enabled));
                }
            case "Release":
                {
                    let ____cat: any = message.Payload['cat'];
                    let ____dog: Dog = message.Payload['dog'];
                    return $UniRpc_ReturnMessage(message, await this.Release(____cat, ____dog));
                }
            case "GetMessage":
                {
                    return $UniRpc_ReturnMessage(message, await this.GetMessage());
                }
            case "IncomingCall":
                {
                    await this.IncomingCall();
                    break;
                }
        }
        throw `${message.Service}.${message.Method} is not defined.`;
    }
}

File: pywebsocket/MyService/DevService.py
from abc import ABC, abstractmethod
from MyService.Message import Message as ImportedRef__MyService_Message
from typing import List, TypeVar
from UniRpc.BaseMessage import BaseMessage as ImportedRef__UniRpc_BaseMessage
from UniRpc.ReturnMessage import ReturnMessage as ImportedRef__UniRpc_ReturnMessage
from UniRpc.WebsocketServiceBase import WebsocketServiceBase as ImportedRef__UniRpc_WebsocketServiceBase

Dog = TypeVar('Dog')
Method_TypeArg__Release__Cat = TypeVar('Method_TypeArg__Release__Cat')

class DevService(ImportedRef__UniRpc_WebsocketServiceBase):
    def __init__(self):
        super().__init__()
        self.__reflection = 'MyService.DevService'

    @abstractmethod
    async def Send(self, message: ImportedRef__MyService_Message) -> bool:
        pass

    @abstractmethod
    async def Get(self) -> bool:
        pass

    @abstractmethod
    async def StoryCount(self) -> float:
        pass

    @abstractmethod
    async def Test(self) -> str:
        pass

    @abstractmethod
    async def AB(self, my: float) -> float:
        pass

    @abstractmethod
    async def Gap(self, value: int, enabled: bool) -> List[int]:
        pass

    @abstractmethod
    async def Release(self, cat: Method_TypeArg__Release__Cat, dog: Dog) -> bytes:
        '''
        release resource
        :param cat: the cat instance
        '''
        pass

    @abstractmethod
    async def GetMessage(self) -> ImportedRef__MyService_Message:
        pass

    @abstractmethod
    async def IncomingCall(self) -> None:
        pass

    async def __invoke(self, message: ImportedRef__UniRpc_BaseMessage) -> ImportedRef__UniRpc_BaseMessage:
        if message.Method == 'Send':
            ____message: ImportedRef__MyService_Message = message.Payload['message']
            return ImportedRef__UniRpc_ReturnMessage(message, await self.Send(____message))
        elif message.Method == 'Get':
            return ImportedRef__UniRpc_ReturnMessage(message, await self.Get())
        elif message.Method == 'StoryCount':
            return ImportedRef__UniRpc_ReturnMessage(message, await self.StoryCount())
        elif message.Method == 'Test':
            return ImportedRef__UniRpc_ReturnMessage(message, await self.Test())
        elif message.Method == 'AB':
            ____my: float = message.Payload['my']
            return ImportedRef__UniRpc_ReturnMessage(message, await self.AB(____my))
        elif message.Method == 'Gap':
            ____value: int = message.Payload['value']
            ____enabled: bool = message.Payload['enabled']
            return ImportedRef__UniRpc_ReturnMessage(message, await self.Gap(____value, ____enabled))
        elif message.Method == 'Release':
            ____cat: any = message.Payload['cat']
            ____dog: Dog = message.Payload['dog']
            return ImportedRef__UniRpc_ReturnMessage(message, await self.Release(____cat, ____dog))
        elif message.Method == 'GetMessage':
            return ImportedRef__UniRpc_ReturnMessage(message, await self.GetMessage())
        elif message.Method == 'IncomingCall':
            await self.IncomingCall()
        else:
            raise Exception(f'{message.Service}.{message.Method} is not defined.')